src/main/java/io/mosip/mimoto/service/impl/IssuersServiceImpl.java

```diff
@@ -94,10 +95,29 @@ public IssuerDTO getIssuerConfig(String issuerId) throws ApiNotAccessibleExcepti
 
     @Override
     public CredentialIssuerWellKnownResponse getIssuerWellknown(String issuerId) throws ApiNotAccessibleException, IOException {
-        IssuerDTO issuerConfig = getIssuerConfig(issuerId);
-        String wellknown = issuerConfig.getWellKnownEndpoint();
-        String wellknownJson = restApiClient.getApi(wellknown, String.class);
-        return new Gson().fromJson(wellknownJson, CredentialIssuerWellKnownResponse.class);
+        CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse = new CredentialIssuerWellKnownResponse();
+        IssuersDTO issuersDto = getAllIssuersWithAllFields();
+        Optional<IssuerDTO> issuerConfigResp = issuersDto.getIssuers().stream()
+                .filter(issuer -> issuer.getCredential_issuer().equals(issuerId))
+                .findFirst();
+        if (issuerConfigResp.isPresent()) {
+            IssuerDTO issuerDto = issuerConfigResp.get();
+            credentialIssuerWellKnownResponse = restApiClient.getApi(issuerDto.getWellKnownEndpoint(), CredentialIssuerWellKnownResponse.class);
+            if (credentialIssuerWellKnownResponse == null) {
+                throw new ApiNotAccessibleException();
+            }
+        }
```

Comment:
we should throw exception if issuerConfigResp not present.
```
issuerConfigResp.map(issuerDto -> {
            credentialIssuerWellKnownResponse = restApiClient.getApi(issuerDto.getWellKnownEndpoint(), CredentialIssuerWellKnownResponse.class);
            if (credentialIssuerWellKnownResponse == null) {
                throw new ApiNotAccessibleException();
            }
        }).orElseThrow(() -> new ApiNotAccessibleException());
```

Reply:
Fixed it.